_posts/2022-07-11-changelog-174.markdown

```diff
@@ -0,0 +1,82 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #174"
+date: 2022-07-11 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#8998] Support [rename refactoring](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#rename-refactoring) (`Refactor | Rename` or <kbd>Shift</kbd>+<kbd>F6</kbd>) for macro [metavariables](https://doc.rust-lang.org/reference/macros-by-example.html#metavariables)
+
+  {% include gif-img.html path="/assets/posts/changelog-174/metavariable-rename" w="700px" %}
+
+* [#8850] Detect more cases by `Borrow checker` inspection
+
+  <img src="/assets/posts/changelog-174/more-borrow-checker-annotations.png" alt="More borrow checker annotations" width="700px">
+
+* [#8972] Detect [E0429](https://doc.rust-lang.org/error-index.html#E0429) and [E0431](https://doc.rust-lang.org/error-index.html#E0431) compiler errors
+
+## Performance Improvements
+
+* [#8985] Invalidate fewer caches after changes in macro call bodies.
+  This speeds up completion and highlighting performance when typing in macro calls
+
+* [#8981] Slightly speed up type inference
+
+* [#8978] Don't invalidate caches in dependency crates on changes in workspace code
+
+* [#8977] Don't trigger the indexing mode when typing in non-recursive macro calls
+
+## Fixes
+
+* [#8977] Show completion popup when typing in a macro automatically as in common code.
+  Previously, you had to invoke completion via <kbd>Ctrl</kbd> + <kbd>Space</kbd> manually
+
+* [#8979] Fix false-positive unresolved reference in explicit type-qualified associated member path like `<S as module::Trait>::convert(S)`
+
+* [#8967] Fix method picking with any allowed [self type](https://doc.rust-lang.org/stable/reference/items/associated-items.html?highlight=self#methods) like `Rc`, `Pin`, etc.
+
+* Improvements of procedural macro expansion:
+  * [#8989] Fix explanation messages why a procedural macro call has not been expanded
+  * [#8991] Fix deserialization of messages produced by procedural macro expander in some cases.
+
+  Note, procedural macro expansion is disabled by default for now.
+  To turn it on, enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental features](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features)
+
+## Internal Improvements
+
+* Improvements in new project model reloading approach:
```

Comment:
```suggestion
* Improvements in the new approach to project model reloading:
```
However, it might be still not clear what are the new and old approaches. Should we mention `org.rust.cargo.new.auto.import`?

Reply:
I think it's ok since it's the internal section
The more detailed explanation will be in the release changelog since we've enabled new project model reloading in master